linux/Platform.c

```diff
@@ -620,124 +620,102 @@ bool Platform_getNetworkIO(NetworkIOData* data) {
-   for (unsigned int i = 0; i < nBatteries; i++)
-      free(batteries[i]);
+        const char* entryName = dirEntry->d_name;
 
-   return total;
-}
+        if(String_startsWith(entryName, "BAT")){
+            //Getting total charge for all battery
+            xSnprintf(filePath, sizeof(filePath), "%s/%s/info", batteryPath, entryName);
+            r = xReadfile(filePath, buffer, sizeof(buffer));
+            if (r < 0) {
+                goto end;
+            }
 
-static ACPresence procAcpiCheck(void) {
-   ACPresence isOn = AC_ERROR;
-   const char* power_supplyPath = PROC_POWERSUPPLY_DIR;
-   DIR* dir = opendir(power_supplyPath);
-   if (!dir)
-      return AC_ERROR;
-
-   for (;;) {
-      const struct dirent* dirEntry = readdir(dir);
-      if (!dirEntry)
-         break;
+            buf = buffer;
 
-      const char* entryName = dirEntry->d_name;
+            while ((line = strsep(&buf, "\n")) != NULL) {
+                memset(field, 0, 100*sizeof(char));
+                sscanf(line,"%[^:]:%d",field,&val);
 
-      if (entryName[0] != 'A')
-         continue;
+                //checking for empty field
+                if(field[0] == '\0')
+                    continue;
 
-      char statePath[256];
-      xSnprintf(statePath, sizeof(statePath), "%s/%s/state", power_supplyPath, entryName);
-      FILE* file = fopen(statePath, "r");
-      if (!file) {
-         isOn = AC_ERROR;
-         continue;
-      }
-      char* line = String_readLine(file);
+                if(String_eq(field, "last full capacity")){
+                    totalFull += val;
+                    break;
+                }
+            }
+            //Getting remaining charge for all battery
+            memset(filePath, 0 ,sizeof(filePath));
+            memset(buffer, 0 ,sizeof(buffer));
+            xSnprintf(filePath, sizeof(filePath), "%s/%s/state", batteryPath, entryName);
+            r = xReadfile(filePath, buffer, sizeof(buffer));
+            if (r < 0) {
+                goto end;
+            }
 
-      fclose(file);
+            buf = buffer;
 
-      if (!line)
-         continue;
+            while ((line = strsep(&buf, "\n")) != NULL) {
+                memset(field, 0, 100*sizeof(char));
+                sscanf(line,"%[^:]:%d",field,&val);
 
-      char* isOnline = String_getToken(line, 2);
-      free(line);
+                //checking for empty field
+                if(field[0] == '\0')
+                    continue;
 
-      if (String_eq(isOnline, "on-line"))
-         isOn = AC_PRESENT;
-      else
-         isOn = AC_ABSENT;
-      free(isOnline);
-      if (isOn == AC_PRESENT)
-         break;
-   }
-
-   closedir(dir);
+                if(String_eq(field, "remaining capacity")){
+                    totalRemain += val;
+                    break;
+                }
+            }
 
-   return isOn;
+        }
+    }
+end:
+    closedir(batteryDir);
+    return (totalFull > 0 ? ((double) totalRemain * 100.0) / (double) totalFull : NAN);
```

Comment:
Avoid unnecessary parens
```suggestion
    return totalFull > 0 ? ((double) totalRemain * 100.0) / (double) totalFull : NAN;
```